lib/src/pie/PieLabel.tsx

```diff
@@ -0,0 +1,61 @@
+import { Text, type Color, type SkFont } from "@shopify/react-native-skia";
+import React, { type ReactNode } from "react";
+import { usePieSliceContext } from "./contexts/PieSliceContext";
+
+type LabelPostion = {
+  x: number;
+  y: number;
+  midAngle: number;
+};
+
+export type PieLabelProps = {
+  font?: SkFont | null;
+  radiusOffset?: number;
+  color?: Color;
+  text?: string;
+  children?: (position: LabelPostion) => ReactNode;
+};
+
+const PieLabel = ({
+  font,
+  radiusOffset = 0.5,
+  color = "white",
+  text,
+  children,
+}: PieLabelProps) => {
+  const { slice } = usePieSliceContext();
+  const labelText = text ?? slice.label;
+
+  const labelWidth =
+    font
+      ?.getGlyphWidths(font.getGlyphIDs(labelText))
+      .reduce((sum, value) => sum + value, 0) ?? 0;
```

Comment:
Perhaps as a clean up after this PR, we should make some kind of internal helper function that does this calculation for us instead of copy-pasting it throughout the codebase 😁